Lib/test/support/__init__.py

```diff
@@ -2070,21 +2072,48 @@ def _is_full_match_test(pattern):
     return ('.' in pattern) and (not re.search(r'[?*\[\]]', pattern))
 
 
-def set_match_tests(patterns):
-    global _match_test_func, _match_test_patterns
+def set_match_tests(accept_patterns, reject_patterns):
+    global _match_test_func, _accept_test_patterns, _ignore_test_patterns
 
-    if patterns == _match_test_patterns:
-        # No change: no need to recompile patterns.
-        return
 
+    if accept_patterns is None:
+        accept_patterns = ()
+    if reject_patterns is None:
+        reject_patterns = ()
+
+    accept_func = reject_func = None
+
+    if accept_patterns != _accept_test_patterns:
+        accept_patterns, accept_func = _update_match_function(accept_patterns)
+    if reject_patterns != _ignore_test_patterns:
+        reject_patterns, reject_func = _update_match_function(reject_patterns)
+
+    # Create a copy since patterns can be mutable and so modified later
+    _accept_test_patterns = tuple(accept_patterns)
+    _ignore_test_patterns = tuple(reject_patterns)
+
+    if accept_func or reject_func:
```

Comment:
I prefer to explicitly check for None:

```suggestion
    if accept_func is not None or reject_func is not None:
```